packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -544,6 +547,84 @@ Generated at ${createdAt}
+   * present in the OpenAPI definition `info` object.
+   *
+   * Additionally if a license in `info.license` is recognized by the SPDX we'll also create a
+   * `LICENSE` file in for their generated SDK. We're only supporting SPDX-recognized licenses for
+   * this because we otherwise have no idea what the license represents.
+   *
+   * @see {@link https://spdx.org/licenses/}
+   */
+  private createSDKHeading() {
+    let docblock: OptionalKind<JSDocStructure> = {};
+
+    // The OpenAPI 3.1 `info` object added some properties that aren't available in 3.0 but 3.1
+    // still contains everything that was available in 3.0 so to allow us to not have to do gross
+    // `as OpenAPIV3_1.InfoObject` typing everywhere we're just assuming that we have 3.1 object
+    // here.
+    const infoObject = this.spec.api.info as OpenAPIV3_1.InfoObject;
+    if (infoObject?.title || infoObject?.description) {
+      docblock.description = writer => {
+        if (infoObject.title) {
+          writer.writeLine(docblockEscape(wordWrap(infoObject.title)));
+        }
+
+        if (infoObject.description) {
+          writer.conditionalNewLine(!!infoObject.title);
+          writer.writeLine(docblockEscape(wordWrap(infoObject.description)));
+        }
+
+        writer.newLineIfLastNot();
+        return writer;
+      };
+    }
+
+    if (infoObject?.termsOfService) {
+      docblock = TSGenerator.#addTagToDocblock(docblock, {
+        tagName: 'see',
+        text: `{@link ${infoObject.termsOfService} Terms of Service}`,
+      });
+    }
+
+    if (infoObject?.license?.name || infoObject?.license?.identifier) {
+      let spdxLicense;
+      if (infoObject?.license?.identifier) {
+        spdxLicense = infoObject?.license?.identifier;
+      } else {
+        // Though `name` is required by the OpenAPI specification but most people seem to use `name`
+        // instead `identifier` for their licensing identifier so if they've done this we should
+        // try to make the license name recognizable by the `license` package. For example this'll
+        // change "Apache 2.0" to the SPDX-recognized "Apache-2.0".
+        spdxLicense = infoObject?.license?.name.replace(/ /g, '-');
+      }
+
+      // If the license they've got has too many matches we shouldn't try to pick a license because
+      // we'll run the risk of licensing it under a license they didn't intend.
+      if (findLicense(spdxLicense).length === 1) {
+        docblock = TSGenerator.#addTagToDocblock(docblock, {
+          tagName: 'license',
+          text: spdxLicense,
+        });
+
+        if (infoObject?.license?.url) {
+          docblock = TSGenerator.#addTagToDocblock(docblock, {
+            tagName: 'see',
+            text: `{@link ${infoObject.license.url}}`,
+          });
+        }
+
+        // We should add a LICENSE file to their SDK too.
+        this.project.createSourceFile('LICENSE', getLicense(spdxLicense));
```

Comment:
Also wondering if we should set a `license` property in the `package.json` file at the same time.